src/coordinator/ts/m3db/convert.go

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package m3db
+
+import (
+	"io"
+	"time"
+
+	"github.com/m3db/m3db/src/dbnode/encoding"
+	"github.com/m3db/m3db/src/dbnode/encoding/m3tsz"
+	"github.com/m3db/m3db/src/dbnode/x/xio"
+)
+
+// BlockReplica contains the replicas for a single m3db block
+type BlockReplica struct {
+	Start     time.Time
+	BlockSize time.Duration
+	Replicas  []encoding.MultiReaderIterator
+}
+
+func convertM3DBSeriesIterators(iterators encoding.SeriesIterators) ([]SeriesBlocks, error) {
+	iterAlloc := func(r io.Reader) encoding.ReaderIterator {
```

Comment:
`defer iterators.Close()`... Although may have to clone ID, Namespace, and Tags when you build each new SeriesIterator (also make sure you use the cloned version when you build a SeriesBlocks 

I think the rest should be safe, but double check with @prateek  to confirm